### PR TITLE
Revise GitHub CI

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:
- add job to check unused deps
- bump actions
- remove unused deps
- use latest OTP/Elixir for linting related jobs

### CODE REVIEW

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -34,13 +67,15 @@ jobs:
...
             _build
-          key: ${{ runner.os }}-mix-${{matrix.otp}}-${{matrix.elixir}}-${{ hashFiles('**/mix.lock') }}
+          key: ${{ runner.os }}-mix-${{matrix.pair.otp}}-${{matrix.pair.elixir}}-${{ hashFiles('**/mix.lock') }}
           restore-keys: |
-            ${{ runner.os }}-mix-${{matrix.otp}}-${{matrix.elixir}}-
+            ${{ runner.os }}-mix-${{matrix.pair.otp}}-${{matrix.pair.elixir}}-
       - run: mix deps.get
       - run: mix compile
+      - run: mix deps.unlock --check-unused
+        if: ${{matrix.lint}} # Only check formatting with the latest verison
```
Comment (kianmeng) [inline .github/workflows/on-push.yml:76]:
```
Run mix deps.unlock --check-unused
** (Mix) Unused dependencies in mix.lock file:

  * :dialyze
  * :earmark
  * :httpotion
  * :ibrowse
  * :poison
```

File: .github/workflows/on-push.yml [hunk 2]
```diff
@@ -7,25 +7,58 @@ env:
...
+      - uses: actions/checkout@v4
       - uses: erlef/setup-beam@v1
         with:
-          otp-version: ${{matrix.otp}}
-          elixir-version: ${{matrix.elixir}}
-      - uses: rrainn/dynamodb-action@v2.0.0
+          otp-version: ${{matrix.pair.otp}}
+          elixir-version: ${{matrix.pair.elixir}}
+      - uses: rrainn/dynamodb-action@v3.0.0
         with:
```
Comment (kianmeng) [inline .github/workflows/on-push.yml:62]:
```
The following actions uses node12 which is deprecated and will be forced to run on
node16: actions/checkout@v2, rrainn/dynamodb-action@v2.0.0. For more info: 
https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/
```

**Comment by bernardd** [comment]
Very nice, thanks heaps @kianmeng! Apologies for the delay in getting to this.

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**